src/ui/shared/WindowContext.ts

```diff
@@ -0,0 +1,50 @@
+export type WindowType = 'popup' | 'tab' | 'dialog';
+export type WindowLayout = 'column' | 'page';
+
+export enum WindowParam {
+  windowType = 'windowType',
+  windowLayout = 'windowLayout',
+  mode = 'mode',
+}
+
+function getParam<T>(param: WindowParam) {
```

Comment:
Maybe we can move setParam logic here too?

Reply:
Ah, I think I got the idea. It would be nice to have something well-typed that we can use instead of manually setting search parameters like this:
```
popupUrl.searchParams.append(WindowParam.windowType, 'tab');
popupUrl.searchParams.append(WindowParam.mode, 'onboarding');
```
Let me think...

Reply:
I came up with a better design, check it out

Reply:
I came up with even better design :) 
Now we have single function with the following signature:
```
export function setUrlContext(
  searchParams: URLSearchParams,
  params: Partial<UrlContext>
)
```

Reply:
Not relevant anymore